### ISSUE TITLE
extended/test_bios_update.robot fails with "Error -1 erasing chip"

### ISSUE THREAD

**Issue by micwalsh** [openbmc/openbmc#848]
I ran extended/test_bios_update.robot against barp01 system.  It failed...

Nov 28 19:37:22 barreleye flash_bios.exe[735]: Starting flasher: bios,/tmp/barreleye.pnor,/org/openbmc/control/flash/bios,
Nov 28 19:37:22 barreleye flash_bios.exe[735]: flasher: /tmp/barreleye.pnor, BMC = 1, address = 0x0
Nov 28 19:37:22 barreleye flash_bios.exe[735]: Setting up flash
Nov 28 19:37:22 barreleye flash_bios.exe[735]: Erasing... (may take a while !) Error -1 erasing chip

[bare-01.test_bios_update.161129.152224.status.txt](https://github.com/openbmc/openbmc/files/620320/bare-01.test_bios_update.161129.152224.status.txt)

**Comment by micwalsh** [comment]
Also, I should mention that doing the code update manually as follows worked fine:
pflash -E -f -i -p /tmp/barreleye.pnor

[log.txt](https://github.com/openbmc/openbmc/files/620372/log.txt)

**Comment by shenki** [comment]
Your bug report must contain the OpenBMC commit that you reproduced the issue with. This can be found in the outout of `/etc/os-release`.

**Comment by geissonator** [comment]
Mike was pinging me on this earlier so I had the info.  Fairly recent build.  Seems like our REST api path to update PNOR is broke but pflash still works fine. 

```
[barreleye]# cat /etc/os-release 
ID="openbmc-phosphor"
NAME="Phosphor OpenBMC (Phosphor OpenBMC Project Reference Distro)"
VERSION="v1.99.0-296"
VERSION_ID="v1.99.0-296-ge743b5a-dirty"
PRETTY_NAME="Phosphor OpenBMC (Phosphor OpenBMC Project Reference Distro) v1.99.0-296"
BUILD_ID="v1.99.0"
```

**Comment by spinler** [comment]
@micwalsh   Your testcases don't collect BMC FFDC on fails?  Maybe I just don't see it?

**Comment by spinler** [comment]
I'm captured the journal, but there isn't much else interesting other than what Mike first posted.

[journal_beye6.txt](https://github.com/openbmc/openbmc/files/625777/journal_beye6.txt)

**Comment by spinler** [comment]
@shenki  How do we figure out why this call is failing?
```
	rc = arch_flash_erase_chip(bl);
	if(rc) {
		fprintf(stderr, "Error %d erasing chip\n", rc);
		return(rc);
	}
```
It recreates every time with Mike's testcase.  Is there any extra data I can capture?

**Comment by geissonator** [comment]
@cyrilbur-ibm any thoughts on this one?  Seems like you've been the one recently in this area :)

**Comment by shenki** [comment]
```
Nov 28 20:03:23 barreleye flash_bios.exe[735]: Starting flasher: bios,/tmp/barreleye.pnor,/org/openbmc/control/flash/bios,
Nov 28 20:03:23 barreleye flash_bios.exe[735]: flasher: /tmp/barreleye.pnor, BMC = 1, address = 0x0
Nov 28 20:03:23 barreleye flash_bios.exe[735]: Setting up flash
Nov 28 20:03:23 barreleye flash_bios.exe[735]: Erasing... (may take a while !) Error -1 erasing chip
```

We should add more debug output to flasher, such as what it passes to the `arch_flash_init` call, and what the return value of `arch_flash_init` is.

**Comment by shenki** [comment]
```
uint8_t
flash(FlashControl* flash_control, enum flash_access chip, uint32_t address, char* write_file, char* obj_path) 
{
        int rc;
        printf("flasher: %s, BMC = %d, address = 0x%x\n", write_file, chip, address); 

        /* Prepare for access */
        rc = flash_access_setup(chip);
        if(rc) {
                return FLASH_SETUP_ERROR;
        }
```

@cyrilbur-ibm thinks that this is the wrong function to be calling, and we need to call `arch_flash_init`.

That does not explain why it ever worked though.

**Comment by cyrilbur-ibm** [comment]
https://github.com/open-power/skiboot/blob/master/external/common/arch_flash_arm.c#L300

I'll send a patch to the skeleton if I can work it out how to gerrit.

Done.

**Comment by cyrilbur-ibm** [comment]
https://gerrit.openbmc-project.xyz/#/c/1452/

**Comment by shenki** [comment]
This is merged as of 1da3a353e58e6439f5b7c5a0634e1b382bb27bfc. Please re-test with `v1.99.0-322-g1da3a353e58e` or later.
